scripts/run_mcmc.py

```diff
@@ -0,0 +1,295 @@
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def get_mutation_matrix(data_fp: str) -> MutationMatrix:
+    """Load the mutation matrix from file.
+
+    Args:
+        data_fp: str
+            Filepath of the data.
+
+    Returns:
+        mut_mat: MutationMatrix
+            observed mutation matrix
+    """
+
+    # load data from file to json object
+    with open(data_fp, "r") as f:
+        data = json.load(f)
+
+    # convert json object to mutation matrix
+    mut_mat = data["noisy_mutation_mat"]  # TODO: adjust to flexible title
+    # convert to array
+    mut_mat = jnp.array(mut_mat)
+
+    return mut_mat
+
+
+def run_chain(params: argparse.Namespace, config: dict, **kwargs) -> None:
+    """Run the MCMC sampler for tree inference.
+
+    Args:
+        params: argparse.Namespace
+            input parameters from parser for MCMC sampler
+        config: dict
+            config dictionary
+
+    Returns:
+        None
+    """
+
+    # Set random seed
+    rng = random.PRNGKey(params.seed)
+
+    # load observed mutation matrix data from file
+    mut_mat = jnp.array(get_mutation_matrix(params.data_fp))
+
+    # check if init tree is provided
+    if params.init_tree_fp is None:
+        # infer dimensions of tree from data
+        n_mutations, m_cells = mut_mat.shape
+        # split rng key
+        rng_tree, rng = random.split(rng, 2)
+        #  generate random trees (uniform sampling) as adjacency matrix
+        #  / add +1 for root
+        tree = tree_inf.generate_random_tree(rng_tree, n_nodes=n_mutations + 1)
+        tree = jnp.array(tree)
+        # make Tree
+        labels = jnp.arange(n_mutations + 1)
+        init_tree = tree_inf.Tree(tree, labels)
+        logging.info("Generated random tree.")
+
+    else:
+        # parse tree from file given as input
+
+        # make path and check if Path exists
+        p = Path(params.init_tree_fp)
+        if not p.exists():
+            raise FileNotFoundError(f"File {params.init_tree_fp} does not exist.")
+
+        if params.init_TreeNode:
+            init_tree_node = serialize.read_tree_node(params.init_tree_fp)
+            # convert TreeNode to Tree
+            init_tree = tree_inf.tree_from_tree_node(init_tree_node)
+            logging.info("Loaded tree (TreeNode) from file.")
+
+        elif params.init_tree_mcmc_no is not None:
+            mcmc_sample = serialize.read_mcmc_samples(params.init_tree_fp)[
+                params.init_tree_mcmc_no - 1
+            ]
+            _, init_tree, _ = tree_inf.unpack_sample(mcmc_sample)
+            logging.info("Loaded tree (mcmc sample) from file.")
+        else:
+            raise ValueError(
+                "Please provide either TreeNode or mcmc sample number,"
+                " to read in tree from file."
+            )
+
+    # Make Move Probabilities
+    prune_and_reattach = config["move_probs"]["prune_and_reattach"]
```

Comment:
Nice that you have it! I think we'll actually use this option quite a lot in the next sprint :slightly_smiling_face: